kernel/vmemitcodes.ml

```diff
@@ -300,14 +300,23 @@ let check_prim_op = function
   | Float64ldshiftexp -> opCHECKLDSHIFTEXP
   | Float64next_up    -> opCHECKNEXTUPFLOAT
   | Float64next_down  -> opCHECKNEXTDOWNFLOAT
-  | Arraymake | Arrayget | Arrayset | Arraydefault | Arraycopy | Arraylength ->
-    assert false
+  | Arraymake | Arrayget | Arrayset | Arraydefault | Arraycopy | Arraylength
+  | Charcode | Charchr
+  | Stringmake | Stringlength | Stringget | Stringsub | Stringcat
+  | Stringequal | Stringlt | Stringle
+    -> assert false
 
 let check_caml_prim_op = function
 | CAML_Arraymake -> opCHECKCAMLCALL2_1
 | CAML_Arrayget -> opCHECKCAMLCALL2
 | CAML_Arrayset -> opCHECKCAMLCALL3_1
 | CAML_Arraydefault | CAML_Arraycopy | CAML_Arraylength -> opCHECKCAMLCALL1
+| CAML_Charcode | CAML_Charchr -> opCHECKCAMLCALL1
+| CAML_Stringmake -> opCHECKCAMLCALL2_1
+| CAML_Stringlength -> opCHECKCAMLCALL1
+| CAML_Stringget -> opCHECKCAMLCALL2
+| CAML_Stringsub -> opCHECKCAMLCALL3_1
```

Comment:
Same remark here. So, `opCHECKCAMLCALL3` (which you might need to add since it does not seem it was ever needed for arrays).

Reply:
I added it, but I'm not sure about:
```diff
diff --git a/kernel/genOpcodeFiles.ml b/kernel/genOpcodeFiles.ml
index 40a86966ad..49347867a9 100644
--- a/kernel/genOpcodeFiles.ml
+++ b/kernel/genOpcodeFiles.ml
@@ -154,6 +154,7 @@ let opcodes =
     "CHECKCAMLCALL2_1", 2;
     "CHECKCAMLCALL1", 2;
     "CHECKCAMLCALL2", 2;
+    "CHECKCAMLCALL3", 2;
     "CHECKCAMLCALL3_1", 2;
     "STOP", 0
   |]
```
Does 2 need to be 3 here? Why do `CHECKCAMLCALL1` and `CHECKCAMLCALL2` both have arity 2?

Reply:
This is the arity of the instruction from the point of view of the interpreter, that is, the length of the instruction (minus 1). This is orthogonal to the actual semantic arity of the instruction. So, 2 is correct here.